### PR TITLE
API for image management

### PR DESCRIPTION
**Pull request by nadvornik**
## What does this PR change?

Add new API calls and new fields in image lists.
Add image upload controller.

Image upload is compatible with the future HTTP API:
```
curl -H "Content-Type: application/json" -c cookies   -d "{'login':'admin','password':'admin'}" https://srv.tf.local/rhn/manager/api/login
curl -H "Content-Type: multipart/form-data" -F "data=@POS_Image_JeOS7_uyuni.x86_64-7.0.0-build87.tar.xz" -c cookies -b cookies https://srv.tf.local/rhn/manager/upload/image
```
## GUI diff

No difference.

- [x] **DONE**

## Documentation
- TBD

- [ ] **DONE**

## Test coverage
- Unit tests were added

- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/16645
Tracks # **add downstream PR, if any**

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: java/code/src/com/redhat/rhn/frontend/security/PxtAuthenticationService.java [hunk 1]
```diff
@@ -73,6 +73,7 @@
         set.add("/rhn/common/DownloadFile");
         // search (safe to be unprotected, since it has no modifying side-effects)
         set.add("/rhn/Search.do");
+        set.add("/rhn/manager/upload/image");
```
Comment (rjmateus) [inline java/code/src/com/redhat/rhn/frontend/security/PxtAuthenticationService.java:76]: do you want to allow unprotrected download the image? don't we need up have authentication on this one?

**Review by rjmateus — APPROVED** [review]
LGTM. just one small question

File: java/code/src/com/redhat/rhn/frontend/security/PxtAuthenticationService.java (hunk 1 above)
Comment (nadvornik) [reply java/code/src/com/redhat/rhn/frontend/security/PxtAuthenticationService.java:76]: If I understand it correctly, it disables only CSRF token checking. The authentication is still enabled by `withImageAdmin` in the controller.

The HTTP API PR does the same:
https://github.com/uyuni-project/uyuni/pull/4910/files#diff-1d51cadaefa85c656ae781cea467ff6cd196d90d322108b18a15004d05a96252R76

Comment (rjmateus) [reply java/code/src/com/redhat/rhn/frontend/security/PxtAuthenticationService.java:76]: you are completely correct. sorry for the noise

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/image/ImageInfoHandler.java [hunk 2]
```diff
@@ -175,6 +260,132 @@ else if (StringUtils.isEmpty(storeLabel)) {
...
+     * @xmlrpc.param #param_desc("string", "arch", image architecture")
+     * @xmlrpc.returntype #param_desc("int", "id", "ID of the image")
+     */
+    public Long importOSImage(User loggedInUser, String name, String version, String arch) {
+        ensureImageAdmin(loggedInUser);
+        if (StringUtils.isEmpty(name)) {
+            throw new InvalidParameterException("Image name cannot be empty.");
+        }
+        ImageStore store = ImageStoreFactory.lookupBylabelAndOrg("SUSE Manager OS Image Store", loggedInUser.getOrg())
+                        .orElseThrow(NoSuchImageStoreException::new);
```
Comment (aaannz) [inline java/code/src/com/redhat/rhn/frontend/xmlrpc/image/ImageInfoHandler.java:285]: Not that there is something to do, but I find this quite fragile. IMO lookup by store type label `os_image` would survive if someone will try to do some deep SUMA/Uyuni renames.

**Review by aaannz — APPROVED** [review]
LGTM, just one "rant" and one comment nitpick.

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/image/ImageInfoHandler.java (hunk 2 above)
Comment (nadvornik) [reply java/code/src/com/redhat/rhn/frontend/xmlrpc/image/ImageInfoHandler.java:285]: The same code is used in unit tests so it can't break silently.

**Pull request merged**